osu.Desktop/OsuGameDesktop.cs

```diff
@@ -146,9 +159,50 @@ protected override string LocateBasePath()
                 return null;
             }
 
-            public StableStorage(DesktopGameHost host)
-                : base(string.Empty, host)
+            /// <summary>
+            /// osu! stable Songs folder can be moved to another location by changing the 'BeatmapDirectory' setting in the stable configuration file.
+            /// This locates the Songs folder location for this stable instalation and sets flags if the songs folder location has been customized.
+            /// </summary>
+            /// <returns></returns>
+            private string locateSongsDirectory()
             {
+                //we get the user config file.
+                var configFile = GetStream(GetFiles(".", "osu!.*.cfg").First());
+                var textReader = new StreamReader(configFile);
+
+                while (!textReader.EndOfStream)
+                {
+                    var line = textReader.ReadLine();
+
+                    if (line.StartsWith("BeatmapDirectory"))
+                    {
+                        var dir = line.Split('=')[1].TrimStart();
+                        if (!dir.Equals(default_songs_path, StringComparison.Ordinal))
+                        {
+                            usesCustomSongsPath = true;
+                            return Path.GetFullPath(dir);
+                        }
```

Comment:
The comparison seems unnecessary, can just assign.

Also you should break out of the loop after reading this line to avoid reading the entire file unnecessarily.